spring-cloud-stream/src/main/java/org/springframework/cloud/stream/error/BinderErrorConfigurer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.error;
+
+
+import org.springframework.cloud.stream.binder.AbstractBinder;
+import org.springframework.cloud.stream.binder.Binding;
+import org.springframework.cloud.stream.binder.ConsumerProperties;
+
+/**
+ * Binder implementations should implement their Error Handling strategy using sub classes of this interface.
+ * The {@link AbstractBinder} uses lifecycle hooks to call each of the methods describe in this interface.
+ * @author Vinicius Carvalho
+ *
+ * @since 1.3
+ */
+public interface BinderErrorConfigurer<C extends ConsumerProperties, T> {
```

Comment:
We may have producer side error configurer once https://github.com/spring-cloud/spring-cloud-stream/issues/916 is addressed. May be we could create a common `BinderErrorConfigurer` that is not specific to `ConsumerProperties` ?

Also, to avoid running into issues like https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/58, we could potentially have both Producer/Consumer properties for the `ErrorConfigurer` that knows how to use Consumer/Producer properties based on the context.